From 7c1765eb24d198ad76a055a8f7d2cd885122f02d Mon Sep 17 00:00:00 2001
From: Sachin Mokashi <sachinx.mokashi@intel.com>
Date: Thu, 5 Oct 2017 13:28:07 +0800
Subject: [PATCH 40/43] x86/platform/apl: Enable spi-pxa2xx auto switch between
 PIO and DMA mode

Add and configure pio_dma_threshold in board configuration for SPI PXA2XX
controller. The driver will automatically switch the data transfer between
PIO and DMA mode depending on the current data transfer length.

Signed-off-by: Sachin Mokashi <sachinx.mokashi@intel.com>
Signed-off-by: Wan Ahmad Zainie <wan.ahmad.zainie.wan.mohamad@intel.com>
---
 arch/x86/platform/apl/apl-board.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/x86/platform/apl/apl-board.c b/arch/x86/platform/apl/apl-board.c
index 55a2770..7b81664 100644
--- a/arch/x86/platform/apl/apl-board.c
+++ b/arch/x86/platform/apl/apl-board.c
@@ -25,6 +25,7 @@
 static struct pxa2xx_spi_chip chip_data = {
 	.gpio_cs = -EINVAL,
 	.dma_burst_size = 1,
+	.pio_dma_threshold = 8,
 };
 
 static struct spi_board_info apl_spi_slaves[] = {
-- 
1.9.1

